src/hiveboard_bridge/src/MessageHandler.cpp

```diff
@@ -4,14 +4,20 @@ MessageHandler::MessageHandler() {}
 
 MessageHandler::~MessageHandler() {}
 
-bool MessageHandler::handleMessage(MessageDTO message) {
+MessageDTO MessageHandler::handleMessage(MessageDTO message) {
+    GenericResponseStatusDTO responseStatus = GenericResponseStatusDTO::BadRequest;
+    uint32_t msgSourceId = message.getSourceId();
+    uint32_t msgDestinationId = message.getDestinationId();
+    uint32_t requestId = 0;
```

Comment:
Shouldn't source and destination be flipped between the request and the response? Otherwise, you're sending the response to yourself